.changeset/strange-wolves-knock.md

```diff
@@ -0,0 +1,24 @@
+---
+'@graphql-mesh/fusion-runtime': minor
+'@graphql-tools/federation': minor
+'@graphql-hive/gateway-runtime': minor
+---
+
+Support promises in `progressiveOverride` option
+
+```ts
+defineConfig({
```

Comment:
We should always think that our readers don't know how to configure the gateway.
```suggestion
import { defineConfig } from '@graphql-hive/gateway';
export const gatewayConfig = defineConfig({
```